### ISSUE TITLE
Test errors with PHP7 

### ISSUE THREAD

**Issue by dominch** [wcgallego/pecl-gearman#18]
Hello, I'm trying to get gearman extension working with PHP7: 

```
root@devGlobal:/root/gearman/pecl-gearman# php -v
PHP 7.0.4-7+deb.sury.org~trusty+2 (cli) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies
```

(I had same issues with 7.0.4-6) 

Everything is ok for those steps: 

```
root@devGlobal:/root/gearman/pecl-gearman# phpize
Configuring for:
PHP Api Version:         20151012
Zend Module Api No:      20151012
Zend Extension Api No:   320151012


root@devGlobal:/root/gearman/pecl-gearman# ./configure
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for a sed that does not truncate output... /bin/sed
checking for cc... cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether cc accepts -g... yes
checking for cc option to accept ISO C89... none needed
checking how to run the C preprocessor... cc -E
checking for icc... no
checking for suncc... no
checking whether cc understands -c and -o together... yes
checking for system library directory... lib
checking if compiler supports -R... no
checking if compiler supports -Wl,-rpath,... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking target system type... x86_64-unknown-linux-gnu
checking for PHP prefix... /usr
checking for PHP includes... -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib
checking for PHP extension directory... /usr/lib/php/20151012
checking for PHP installed headers prefix... /usr/include/php/20151012
checking if debug is enabled... no
checking if zts is enabled... no
checking for re2c... no
configure: WARNING: You will need re2c 0.13.4 or later if you want to regenerate PHP parsers.
checking for gawk... gawk
checking whether to enable gearman support... yes, shared
found in /usr
checking for gearman_client_set_context in -lgearman... yes
checking for gearman_worker_set_server_option in -lgearman... yes
checking for gearman_job_error in -lgearman... yes
checking for gearman_client_unique_status in -lgearman... yes
checking how to print strings... printf
checking for a sed that does not truncate output... (cached) /bin/sed
checking for fgrep... /bin/grep -F
checking for ld used by cc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking for gawk... (cached) gawk
checking command to parse /usr/bin/nm -B output from cc object... ok
checking for sysroot... no
checking for mt... mt
checking if mt is a manifest tool... no
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if cc supports -fno-rtti -fno-exceptions... no
checking for cc option to produce PIC... -fPIC -DPIC
checking if cc PIC flag -fPIC -DPIC works... yes
checking if cc static flag -static works... yes
checking if cc supports -c -o file.o... yes
checking if cc supports -c -o file.o... (cached) yes
checking whether the cc linker (/usr/bin/ld -m elf_x86_64) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
configure: creating ./config.status
config.status: creating config.h
config.status: config.h is unchanged



root@devGlobal:/root/gearman/pecl-gearman# make
/bin/bash /root/gearman/pecl-gearman/libtool --mode=compile cc  -I. -I/root/gearman/pecl-gearman -DPHP_ATOM_INC -I/root/gearman/pecl-gearman/include -I/root/gearman/pecl-gearman/main -I/root/gearman/pecl-gearman -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib  -DHAVE_CONFIG_H  -g -O2 -Wall   -c /root/gearman/pecl-gearman/php_gearman.c -o php_gearman.lo
libtool: compile:  cc -I. -I/root/gearman/pecl-gearman -DPHP_ATOM_INC -I/root/gearman/pecl-gearman/include -I/root/gearman/pecl-gearman/main -I/root/gearman/pecl-gearman -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib -DHAVE_CONFIG_H -g -O2 -Wall -c /root/gearman/pecl-gearman/php_gearman.c  -fPIC -DPIC -o .libs/php_gearman.o
/bin/bash /root/gearman/pecl-gearman/libtool --mode=link cc -DPHP_ATOM_INC -I/root/gearman/pecl-gearman/include -I/root/gearman/pecl-gearman/main -I/root/gearman/pecl-gearman -I/usr/include/php/20151012 -I/usr/include/php/20151012/main -I/usr/include/php/20151012/TSRM -I/usr/include/php/20151012/Zend -I/usr/include/php/20151012/ext -I/usr/include/php/20151012/ext/date/lib  -DHAVE_CONFIG_H  -g -O2 -Wall   -o gearman.la -export-dynamic -avoid-version -prefer-pic -module -rpath /root/gearman/pecl-gearman/modules  php_gearman.lo -lgearman -lgearman -lgearman -lgearman
libtool: link: cc -shared  -fPIC -DPIC  .libs/php_gearman.o   -lgearman  -O2   -Wl,-soname -Wl,gearman.so -o .libs/gearman.so
libtool: link: ( cd ".libs" && rm -f "gearman.la" && ln -s "../gearman.la" "gearman.la" )
/bin/bash /root/gearman/pecl-gearman/libtool --mode=install cp ./gearman.la /root/gearman/pecl-gearman/modules
libtool: install: cp ./.libs/gearman.so /root/gearman/pecl-gearman/modules/gearman.so
libtool: install: cp ./.libs/gearman.lai /root/gearman/pecl-gearman/modules/gearman.la
libtool: finish: PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/sbin" ldconfig -n /root/gearman/pecl-gearman/modules
----------------------------------------------------------------------
Libraries have been installed in:
   /root/gearman/pecl-gearman/modules

If you ever happen to want to link against installed libraries
in a given directory, LIBDIR, you must either use libtool, and
specify the full pathname of the library, or use the `-LLIBDIR'
flag during linking and do at least one of the following:
   - add LIBDIR to the `LD_LIBRARY_PATH' environment variable
     during execution
   - add LIBDIR to the `LD_RUN_PATH' environment variable
     during linking
   - use the `-Wl,-rpath -Wl,LIBDIR' linker flag
   - have your system administrator add LIBDIR to `/etc/ld.so.conf'

See any operating system documentation about shared libraries for
more information, such as the ld(1) and ld.so(8) manual pages.
----------------------------------------------------------------------

Build complete.
Don't forget to run 'make test'.
```

Here I can copy and enable module and it seems to work (I can start worker and pass something there). As soon as worker receive task it gives me segmentation fault and core dump. I decided to run some simpler tests like those in main module directory and they are NOT working :( 

```
root@devGlobal:/root/gearman/pecl-gearman# php test_client.php
# TESTING CLIENT INTERFACE
gearman_client_create() pass
PHP Fatal error:  Uncaught Error: Call to undefined function gearman_client_clone() in /root/gearman/pecl-gearman/test_client.php:27
Stack trace:
#0 {main}
  thrown in /root/gearman/pecl-gearman/test_client.php on line 27
```

This is strange, seems like one function is not registered correctly. Test suite gives me more errors: 

```
root@devGlobal:/root/gearman/pecl-gearman# make test

Build complete.
Don't forget to run 'make test'.


=====================================================================
PHP         : /usr/bin/php7.0
PHP_SAPI    : cli
PHP_VERSION : 7.0.4-7+deb.sury.org~trusty+2
ZEND_VERSION: 3.0.0
PHP_OS      : Linux - Linux devGlobal 3.19.0-51-generic #58~14.04.1-Ubuntu SMP Fri Feb 26 22:02:58 UTC 2016 x86_64
INI actual  : /root/gearman/pecl-gearman/tmp-php.ini
More .INIs  :
CWD         : /root/gearman/pecl-gearman
Extra dirs  :
VALGRIND    : Not used
=====================================================================
TIME START 2016-03-24 20:42:26
=====================================================================
PASS Check for gearman  presence [tests/gearman_001.phpt]
PASS gearman_worker_set_id() [tests/gearman_002.phpt]
PASS new GearmanClient(), GearmanClient::gearman_client_create() [tests/gearman_client_001.phpt]
PASS GearmanClient::returnCode(), gearman_client_return_code() [tests/gearman_client_002.phpt]
PASS GearmanClient::error(), gearman_client_error() [tests/gearman_client_003.phpt]
PASS GearmanClient::getErrno(), gearman_client_get_errno() [tests/gearman_client_004.phpt]
FAIL GearmanClient::options(), gearman_client_options() [tests/gearman_client_005.phpt]
FAIL GearmanClient::setOptions(), gearman_client_set_options() [tests/gearman_client_006.phpt]
PASS GearmanClient::addOptions(), gearman_client_add_options() [tests/gearman_client_007.phpt]
FAIL GearmanClient::removeOptions(), gearman_client_remove_options() [tests/gearman_client_008.phpt]
PASS GearmanClient::timeout(), gearman_client_timeout() [tests/gearman_client_009.phpt]
PASS GearmanClient::setTimeout(), gearman_client_set_timeout() [tests/gearman_client_010.phpt]
PASS GearmanClient::context(), GearmanClient::setContext(), gearman_client_context(), gearman_client_set_context() [tests/gearman_client_011.phpt]
PASS GearmanClient::addServer(), gearman_client_add_server() [tests/gearman_client_012.phpt]
PASS GearmanClient::addServers(), gearman_client_add_servers() [tests/gearman_client_013.phpt]
PASS gearman_worker_return_code() [tests/gearman_worker_001.phpt]
PASS gearman_worker_return_code() [tests/gearman_worker_002.phpt]
PASS gearman_worker_error() [tests/gearman_worker_003.phpt]
PASS gearman_worker_errno() [tests/gearman_worker_004.phpt]
PASS gearman_worker_options() [tests/gearman_worker_005.phpt]
PASS gearman_worker_set_options() [tests/gearman_worker_006.phpt]
PASS gearman_worker_add_options() [tests/gearman_worker_007.phpt]
PASS gearman_worker_timeout(), gearman_worker_set_timeout() [tests/gearman_worker_008.phpt]
PASS gearman_worker_set_id() [tests/gearman_worker_009.phpt]
PASS gearman_worker_add_server(), gearman_worker_add_servers() [tests/gearman_worker_010.phpt]
PASS gearman_worker_wait() [tests/gearman_worker_011.phpt]
PASS gearman_worker_register(), gearman_worker_unregister(), gearman_worker_unregister_all() [tests/gearman_worker_012.phpt]
PASS gearman_worker_add_function() [tests/gearman_worker_013.phpt]
FAIL gearman_worker_work() [tests/gearman_worker_014.phpt]
FAIL gearman_worker_echo() [tests/gearman_worker_015.phpt]
=====================================================================
TIME END 2016-03-24 20:42:27

=====================================================================
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :   14
---------------------------------------------------------------------

Number of tests :   30                30
Tests skipped   :    0 (  0.0%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :    5 ( 16.7%) ( 16.7%)
Expected fail   :    0 (  0.0%) (  0.0%)
Tests passed    :   25 ( 83.3%) ( 83.3%)
---------------------------------------------------------------------
Time taken      :    1 seconds
=====================================================================

=====================================================================
FAILED TEST SUMMARY
---------------------------------------------------------------------
GearmanClient::options(), gearman_client_options() [tests/gearman_client_005.phpt]
GearmanClient::setOptions(), gearman_client_set_options() [tests/gearman_client_006.phpt]
GearmanClient::removeOptions(), gearman_client_remove_options() [tests/gearman_client_008.phpt]
gearman_worker_work() [tests/gearman_worker_014.phpt]
gearman_worker_echo() [tests/gearman_worker_015.phpt]
=====================================================================

You may have found a problem in PHP.
This report can be automatically sent to the PHP QA team at
http://qa.php.net/reports and http://news.php.net/php.qa.reports
This gives us a better understanding of PHP's behavior.
If you don't want to send the report immediately you can choose
option "s" to save it.  You can then email it to qa-reports@lists.php.net later.
Do you want to send this report now? [Yns]:

Please enter your email address.
(Your address will be mangled so that it will not go out on any
mailinglist in plain text):

Posting to http://qa.php.net/buildtest-process.php

Thank you for helping to make PHP better.
```

More information: 

```
root@devGlobal:/root/gearman/pecl-gearman# dpkg -s libgearman-dev
Package: libgearman-dev
Status: install ok installed
Priority: extra
Section: libdevel
Installed-Size: 409
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Source: gearmand
Version: 1.0.6-3
Depends: libgearman7 (= 1.0.6-3), libevent-dev
Description: Development files for the Gearman Library
 Gearman is a system to farm out work to other machines, dispatching function
 calls to machines that are better suited to do work, to do work in parallel,
 to load balance lots of function calls, or to call functions between languages.
 .
 This package contains the development headers for the libgearman library.
Original-Maintainer: Stig Sandbeck Mathisen <ssm@debian.org>
Homepage: http://launchpad.net/gearmand
```

I'm stuck here. Am I doing something wrong that it's not passing tests correctly? Is there anything missing? Of course I submitted report to qa-reports and reported it here. 

Module installed is on php -m list, and seems to report version correctly: 

```
root@devGlobal:/root/gearman/pecl-gearman# php -i | grep gear
/etc/php/7.0/cli/conf.d/20-gearman.ini,
gearman
gearman support => enabled
libgearman version => 1.0.6
OLDPWD => /root/gearman
PWD => /root/gearman/pecl-gearman
$_SERVER['OLDPWD'] => /root/gearman
$_SERVER['PWD'] => /root/gearman/pecl-gearman
```

Like in tests - some functions are missing: 

```
root@devGlobal:/root/gearman/pecl-gearman# php test_client.php
# TESTING CLIENT INTERFACE
gearman_client_create() pass
PHP Fatal error:  Uncaught Error: Call to undefined function gearman_client_clone() in /root/gearman/pecl-gearman/test_client.php:27
Stack trace:
#0 {main}
  thrown in /root/gearman/pecl-gearman/test_client.php on line 27
```

I can attach core dump from it, but I think that problem is clearly visible here. 

ps. of course I'm using latest source from this repo: 

```
root@devGlobal:/root/gearman/pecl-gearman# git log -n 1
commit 9e32a58ca8a33caba7d3b45a49357f6e801e3199
Author: Will Gallego <wcgallego@gmail.com>
Date:   Fri Mar 18 14:09:39 2016 +0000

    GearmanWorker::addServer() optional params

    params should be optional for addServer, where host and port default
    to 'localhost' and 4730 respectively
```

**Comment by tomahock** [comment]
Same here:

```
# TESTING CLIENT INTERFACE
gearman_client_create() pass
PHP Fatal error:  Uncaught Error: Call to undefined function gearman_client_clone() in /servers/sources/pecl-gearman/test_client.php:27
Stack trace:
#0 {main}
  thrown in /servers/sources/pecl-gearman/test_client.php on line 27

Fatal error: Uncaught Error: Call to undefined function gearman_client_clone() in /servers/sources/pecl-gearman/test_client.php:27
Stack trace:
#0 {main}
  thrown in /servers/sources/pecl-gearman/test_client.php on line 27
```

**Comment by wcgallego** [comment]
Hey All,

So 'make test' is what I've been working on as canonical testing for the gearman package. test_client.php hasn't been updated.

As for gearman_client_clone, this hasn't been supported in the code before the migration to PHP 7 and I don't immediately have plans to put it in. Really, the test_client.php should be deprecated.

You do need to have gearmand running to have make test pass. You can see the results of these tests by looking in the tests directory and looking at files that end in '.out'. If you can confirm gearmand is running on the default host/port (localhost:4730) and your instance can talk to it, then rerun make test and paste in the results from .out files, it'll go a long way to addressing this. Thanks!

**Comment by dominch** [comment]
Hello, 

Seems that last two tests fails with json module dependency: 

```
Fatal error: Uncaught Error: Call to undefined function json_encode() in /root/gearman/pecl-gearman/tests/gearman_worker_015.php:8
Stack trace:
#0 {main}
  thrown in /root/gearman/pecl-gearman/tests/gearman_worker_015.php on line 8
```

Seems like it don't load json module, which is normally present: 

```
root@devGlobal:/root/gearman/pecl-gearman/tests# php -m | grep json
json
```

and it works: 

```
root@devGlobal:/root/gearman/pecl-gearman/tests# echo "echo json_encode(['foo' => 'bar']);" | php -a
Interactive mode enabled

{"foo":"bar"}
```

reviewing command which is used I think that problem is with -n switch (to not load any other ini): 

```
root@devGlobal:/root/gearman/pecl-gearman/tests# php -n -m
[PHP Modules]
Core
date
filter
hash
libxml
openssl
pcntl
pcre
Reflection
session
SPL
standard
zlib

[Zend Modules]
```

those two are ok when run without `-n` (no ini) and `-d extension_dir...`, so that will work when module is copied to extension dir. 

For others tests - they seems run without any fatal error, some details are different. Please take a look: 

```
root@devGlobal:/root/gearman/pecl-gearman/tests# for file in *.out; do printf "\n$file\n\n"; cat $file; printf "\n\n"; done

gearman_client_005.out

GearmanClient::options (OO): 96
gearman_client_options (Procedural): 96
OK


gearman_client_006.out

GearmanClient::options (OO): 96
GearmanClient::setOptions (OO): Success
GearmanClient::options (OO): 2
gearman_client_options (Procedural): 96
gearman_client_set_options (Procedural): Success
gearman_client_options (Procedural): 2
OK


gearman_client_008.out

GearmanClient::options (OO): 96
GearmanClient::removeOptions (OO): Success
GearmanClient::options (OO): 64
gearman_client_options (Procedural): 96
gearman_client_remove_options (Procedural): Success
gearman_client_options (Procedural): 64
OK


gearman_worker_014.out

Fatal error: Uncaught Error: Call to undefined function json_encode() in /root/gearman/pecl-gearman/tests/gearman_worker_014.php:7
Stack trace:
#0 {main}
  thrown in /root/gearman/pecl-gearman/tests/gearman_worker_014.php on line 7


gearman_worker_015.out

Fatal error: Uncaught Error: Call to undefined function json_encode() in /root/gearman/pecl-gearman/tests/gearman_worker_015.php:8
Stack trace:
#0 {main}
  thrown in /root/gearman/pecl-gearman/tests/gearman_worker_015.php on line 8

```

I'll add another ticket with core dump about mine example, seems to be something else.

**Comment by wcgallego** [comment]
Hey @dominch 

I figured it was easier to just remove json_encode from the tests and drop in the string that would've been there anyways. So gearman_worker tests 14 and 15 should now run cleanly without it.

As for the gearman client tests, I bet I've assumed that the default options would be the same everywhere. I'll poke at cleaning that up as well and push that change in. Thanks again for helping to clean up the tests! Clearly they're much needed.

**Comment by wcgallego** [comment]
Ok! I've forced setting the default option value before doing the checks. I'd love to avoid using functions as assumptions before setting them, but not sure of a better way yet. Suggestions welcome.

Anyways, please give this a test and see if any others are failing for you.

**Comment by dominch** [comment]
Hello, 
I just updated it and tried, it's OK: 

```
=====================================================================
TIME START 2016-04-04 09:47:31
=====================================================================
PASS Check for gearman  presence [tests/gearman_001.phpt]
PASS gearman_worker_set_id() [tests/gearman_002.phpt]
PASS new GearmanClient(), GearmanClient::gearman_client_create() [tests/gearman_client_001.phpt]
PASS GearmanClient::returnCode(), gearman_client_return_code() [tests/gearman_client_002.phpt]
PASS GearmanClient::error(), gearman_client_error() [tests/gearman_client_003.phpt]
PASS GearmanClient::getErrno(), gearman_client_get_errno() [tests/gearman_client_004.phpt]
PASS GearmanClient::options(), gearman_client_options() [tests/gearman_client_005.phpt]
PASS GearmanClient::setOptions(), gearman_client_set_options() [tests/gearman_client_006.phpt]
PASS GearmanClient::addOptions(), gearman_client_add_options() [tests/gearman_client_007.phpt]
PASS GearmanClient::removeOptions(), gearman_client_remove_options() [tests/gearman_client_008.phpt]
PASS GearmanClient::timeout(), gearman_client_timeout() [tests/gearman_client_009.phpt]
PASS GearmanClient::setTimeout(), gearman_client_set_timeout() [tests/gearman_client_010.phpt]
PASS GearmanClient::context(), GearmanClient::setContext(), gearman_client_context(), gearman_client_set_context() [tests/gearman_client_011.phpt]
PASS GearmanClient::addServer(), gearman_client_add_server() [tests/gearman_client_012.phpt]
PASS GearmanClient::addServers(), gearman_client_add_servers() [tests/gearman_client_013.phpt]
PASS gearman_worker_return_code() [tests/gearman_worker_001.phpt]
PASS gearman_worker_return_code() [tests/gearman_worker_002.phpt]
PASS gearman_worker_error() [tests/gearman_worker_003.phpt]
PASS gearman_worker_errno() [tests/gearman_worker_004.phpt]
PASS gearman_worker_options() [tests/gearman_worker_005.phpt]
PASS gearman_worker_set_options() [tests/gearman_worker_006.phpt]
PASS gearman_worker_add_options() [tests/gearman_worker_007.phpt]
PASS gearman_worker_timeout(), gearman_worker_set_timeout() [tests/gearman_worker_008.phpt]
PASS gearman_worker_set_id() [tests/gearman_worker_009.phpt]
PASS gearman_worker_add_server(), gearman_worker_add_servers() [tests/gearman_worker_010.phpt]
PASS gearman_worker_wait() [tests/gearman_worker_011.phpt]
PASS gearman_worker_register(), gearman_worker_unregister(), gearman_worker_unregister_all() [tests/gearman_worker_012.phpt]
PASS gearman_worker_add_function() [tests/gearman_worker_013.phpt]
PASS gearman_worker_work() [tests/gearman_worker_014.phpt]
PASS gearman_worker_echo() [tests/gearman_worker_015.phpt]
=====================================================================
TIME END 2016-04-04 09:47:32

=====================================================================
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :   14
---------------------------------------------------------------------

Number of tests :   30                30
Tests skipped   :    0 (  0.0%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :    0 (  0.0%) (  0.0%)
Expected fail   :    0 (  0.0%) (  0.0%)
Tests passed    :   30 (100.0%) (100.0%)
---------------------------------------------------------------------
Time taken      :    1 seconds
=====================================================================
```

Thanks for quick response and fixes, sure it's not that important to require json module on tests. :) You will probably hit few other configuration issues in future, let's just wait for suitable ticket :) 
Thanks again :)
